### PR TITLE
[NETBEANS-54] Module Review editor

### PR DESCRIPTION
**Pull request by vieiro**
- No external libraries.
- Checked Rat report: added license to project files. Did NOT add license to editor/demosrc/README nor to the rest of test files.
- Unit tests org.netbeans.editor.PlainDocumentCompatibilityRandomTest and org.netbeans.modules.editor.NbEditorToolBarTest fail (but they were also failing before this PR).
- QA Tests should be run.

### CODE REVIEW

**Comment by junichi11** [comment]
Merged in 215acff0f9714513987031fae8bf329b0514789c.

@vieiro Do you mind closing this PR?

**Comment by vieiro** [comment]
Closing PR as requested.

**Comment by geertjanw** [comment]
What about these, can you exclude them via Rat or license them:

editor/demosrc/base/org/netbeans/editor/example/res/template.html_
editor/demosrc/base/org/netbeans/editor/example/res/template.java_
editor/demosrc/properties-addon/org/netbeans/editor/example/res/template.properties
editor/test/qa-functional/src/org/netbeans/test/editor/suites/keybindings/actions.txt
editor/test/unit/src/org/netbeans/modules/editor/resources/testAnnotation.xml
editor/test/unit/src/org/netbeans/modules/editor/resources/testAnnotation1.xml
editor/test/unit/src/org/netbeans/modules/editor/resources/testAnnotation2.xml
editor/test/unit/src/org/netbeans/modules/editor/resources/testAnnotation3.xml
editor/test/unit/src/org/netbeans/modules/editor/resources/testAnnotation4.xml
editor/test/unit/src/org/netbeans/modules/editor/resources/testAnnotation5.xml
editor/test/unit/src/org/netbeans/modules/editor/resources/testAnnotation6.xml
editor/test/unit/src/org/netbeans/modules/editor/resources/testAnnotation7.xml
editor/test/unit/src/org/netbeans/modules/editor/resources/testAnnotation8.xml
editor/test/unit/src/org/netbeans/modules/editor/resources/testAnnotationBrowseable1.xml
editor/test/unit/src/org/netbeans/modules/editor/resources/testAnnotationBrowseable2.xml
editor/test/unit/src/org/netbeans/modules/editor/resources/testAnnotationBrowseable3.xml
editor/test/unit/src/org/netbeans/modules/editor/resources/testAnnotationPriority1.xml
editor/test/unit/src/org/netbeans/modules/editor/resources/testAnnotationPriority2.xml
editor/test/unit/src/org/netbeans/modules/editor/resources/testAnnotationPriority3.xml
editor/test/unit/src/org/netbeans/modules/editor/resources/testAnnotationPriority4.xml

**Comment by geertjanw** [comment]
Or is there a problem when these files are relicensed, e.g., tests fail?

**Comment by geertjanw** [comment]
Reopening with the above question.

**Comment by vieiro** [comment]
Will take a look at them tonight.

File: nbbuild/build.xml [hunk 1]
```diff
@@ -2002,6 +2002,14 @@ It is possible to use -Ddebug.port=3234 -Ddebug.pause=y to start the system in d
             <exclude name="debugger.jpda/test/unit/src/org/netbeans/api/debugger/jpda/testapps/JspLineBreakpointApp.txt" /> <!-- test data -->
             <exclude name="diff/test/unit/src/org/netbeans/modules/diff/builtin/provider/*.txt" /> <!--test data-->
             <exclude name="diff/test/unit/src/org/netbeans/modules/diff/builtin/visualizer/data/**" /> <!--test data-->
+            <exclude name="editor/test/qa-functional/data/projects/editor_test/src/**.txt" /> <!-- test data -->
+            <exclude name="editor/test/qa-functional/data/**.pass" /> <!-- qa test data -->
```
Comment (junichi11) [inline nbbuild/build.xml]: These two lines are excluded with `<exclude name="*/test/qa-functional/data/**" />`.

File: nbbuild/build.xml [hunk 2]
```diff
@@ -2002,6 +2002,14 @@ It is possible to use -Ddebug.port=3234 -Ddebug.pause=y to start the system in d
...
+            <exclude name="editor/test/qa-functional/src/org/netbeans/test/editor/suites/keybindings/actions.txt" /> <!-- qa test data -->
+            <exclude name="editor/test/qa-functional/data/projects/editor_test/src/org/netbeans/test/editor/search/IncrementalSearchTest/**.txt" /> <!-- qa test data -->
```
Comment (junichi11) [inline nbbuild/build.xml]: This line is excluded with `<exclude name="*/test/qa-functional/data/**" />`.

File: nbbuild/build.xml [hunk 3]
```diff
@@ -2002,6 +2002,14 @@ It is possible to use -Ddebug.port=3234 -Ddebug.pause=y to start the system in d
...
+            <exclude name="editor/test/unit/src/org/netbeans/modules/editor/resources/**.xml" /> <!-- unit test data -->
```
Comment (junichi11) [inline nbbuild/build.xml]: I suggest `<exclude name="editor/test/unit/src/org/netbeans/modules/editor/resources/testAnnotation*.xml" />`.

**Comment by vieiro** [comment]
Thanks, guys, for your reviews. 

I'll do a new PR tonight/EU Time.

**Comment by vieiro** [comment]
See #236 https://github.com/apache/incubator-netbeans/pull/236

@geertjanw I missed your first comments completely (lost in email folders), please open an issue for these updates, I won't miss those!

Closing this PR now.

**Pull request closed without merging**